### PR TITLE
Fixes #215 - Make tree navigateble by link

### LINKED ISSUE #215: Make tree navigatable (jump directly to tree with link)
## Background: Why this issue? 

Currently it is not possible to jump directly to a tree on the map (including zoom in), although there is a link shown in browser bar when clicking on a tree on the map. E.g. this tree in Schönhauser Allee 8 has the link https://www.giessdenkiez.de/search?location=_3lsj2jt06 but calling it directly only shows the complete map.

## Deliverables: How to fix the issue?

Do a programmatic selection of that tree in the map (filtering by id) or use lat, lng and zoom to adjust map to that point.

## Output: Gain of fixing the issue?

You can send the link to other peaple and they can see directly the tree. 
This feature would be also of help in the adopted view of user's profile, that shows the trees only by name, to be able to jump to these trees directly.

## Issue creation checklist
- [ ] I have attached one of the priority labels `HIGH`, `MIDDLE`, `LOW` --> LOW
- [ ] I have attached other labels with regard to the issues purpose (i.e.`enhancement`, `bug`, `community feedback`) --> Enhancement

### PR DESCRIPTION
**Pull request by vogelino**
This pull request:
- refactors the map component to always fetch tree information when the `highlightedObject` state changes.
- refactors the currently names `isLoading` store state to a more explicit `isMapDataLoading`
- adds a state property called `isTreeMapLoading` and sets it to `false` once the Mapbox component is rendered
- adds a `useEffect` hook in the `AppWrapper` component to scan for the `location` URL query parameter and sets the `highlightedObject` state if found. This hook also takes care of setting the `overlay` state to `false` if there is a query parameter `location` or to `true` if there isn't, thus showing/hiding the splash screen accordingly.
- ensures the map's UI elements are not visible while the splash screen is visible
- ensures that the cookie banner is always displayed above the other elements (sidebar, map zoom, etc)

### CODE REVIEW

**Comment by ff6347** [comment]
@vogelino This is awsome!

File: src/components/map/index.tsx [hunk 1]
```diff
@@ -327,45 +326,38 @@ class DeckGLMap extends React.Component {
     }
   }
 
-  _onClick(x, y, object) {
+  async selectTree(treeId: string) {
     const { setViewport } = this.props;
```
Comment (vogelino) [inline src/components/map/index.tsx:330]: Grabbing this from the `Action(store)` initialisation didn't properly work as it seems that the `connect` fuction is doing some magic (applying the current state to the first argument of the `setViewport` function in `Actions.tsx`).
This is why I import `setViewport` from the props and `getTree` from `Actions(store)`. This should be cleaned up in the future.

Comment (ff6347) [reply src/components/map/index.tsx:330]: Okay.

**Comment by vogelino** [comment]
> @vogelino This is awsome!

Merci! 🙏

**Comment by vogelino** [comment]
On thing I'm realising is that I should redirect to the homepage if the id given in the url doesn't exist in the database. For now, the sidebar hungs up with its loading state.

**Comment by vogelino** [comment]
> One thing I'm realising is that I should redirect to the homepage if the id given in the URL doesn't exist in the database. For now, the sidebar hangs up with its loading state.

I have now implemented it in a way that simply ignores the query parameter if it is invalid (e249ad217560175cb4f95d5e1b13b34c5781c2d9). However in this case I don't show the splash screen. I don't know it this is whished but I assume that an invalid id might also come from inside the app and showing the splash screen, in this case, would feel wrong.

**Review by ff6347 — APPROVED** [review]
LGTM

**Pull request merged**